src/ruby/pb/test/xds_client.rb

```diff
@@ -121,25 +150,42 @@ def run_test_loop(stub, target_seconds_between_rpcs, fail_on_failed_rpcs)
       target_next_start += target_seconds_between_rpcs
       sleep(sleep_seconds)
     end
-    begin
-      deadline = GRPC::Core::TimeConsts::from_relative_time(30) # 30 seconds
-      resp = stub.unary_call(req, deadline: deadline)
-      remote_peer = resp.hostname
-    rescue GRPC::BadStatus => e
-      remote_peer = ""
-      GRPC.logger.info("ruby xds: rpc failed:|#{e.message}|, " \
-                       "this may or may not be expected")
-      if fail_on_failed_rpcs
-        raise e
+    deadline = GRPC::Core::TimeConsts::from_relative_time(30) # 30 seconds
+    results = {}
+    rpcs_to_send.each do |rpc|
+      metadata = metadata_to_send.key?(rpc) ? metadata_to_send[rpc] : {}
+      if rpc == 'UnaryCall'
+        op = stub.unary_call(simple_req,
+                             metadata: metadata,
+                             deadline: deadline,
+                             return_op: true)
+      elsif rpc == 'EmptyCall'
+        op = stub.empty_call(empty_req,
+                             metadata: metadata,
+                             deadline: deadline,
+                             return_op: true)
+      else
+        raise "Unsupported rpc %s" % [rpc]
       end
+      results[rpc] = execute_rpc(op, fail_on_failed_rpcs)
     end
     $watchers_mutex.synchronize do
       $watchers.each do |watcher|
+        # this is counted once when each group of all rpcs_to_send were done
```

Comment:
is there a place in the spec that says this? If so can we link this comment to there?

Reply:
The "change" to the spec, which this PR addressed, hasn't been merged to the main xDS test description doc yet. It's just in an internal doc linked from the internal bug b/161801377.

In there, it specified that the new `--rpc` param will specify the list of RPCs to make (i.e. `UnaryCall`, `EmptyCall`, etc). And `--qps=10` means to _make 10 RPCs/second for each type_.